### PR TITLE
[minor] Add get function for parameters

### LINKED ISSUE #252: add model.get_params and set_params
similar to scikit model.get_params

for scenario testing or for parameter-based save / load functions

### PR DESCRIPTION
**Pull request by noxan**
Replacement for #935, rebasing the branch and work from @anastasiia-tovstolis

Close #252

### CODE REVIEW

File: neuralprophet/forecaster.py [hunk 1]
```diff
@@ -340,6 +340,10 @@ def __init__(
         accelerator: Optional[str] = None,
         trainer_config: dict = {},
     ):
+        kwargs = locals()
+        self.config = kwargs
+        self.config.pop("self")
```
Comment (karl-richter) [inline neuralprophet/forecaster.py:345]: What exactly does this line do?

**Review by karl-richter — APPROVED** [review]

**Comment by karl-richter** [comment]
@noxan please check the pyright test

File: neuralprophet/forecaster.py (hunk 1 above)
Comment (noxan) [reply neuralprophet/forecaster.py:345]: it removes the `self` reference from the locals() call, so we do only keep the provided arguments

File: neuralprophet/forecaster.py [hunk 2]
```diff
@@ -340,6 +340,10 @@ def __init__(
         accelerator: Optional[str] = None,
         trainer_config: dict = {},
     ):
+        kwargs = locals()
+        self.config = kwargs
```
Comment (noxan) [inline neuralprophet/forecaster.py:344]:
```suggestion
        self.config = locals()
```

File: neuralprophet/forecaster.py (hunk 1 above)
Comment (noxan) [reply neuralprophet/forecaster.py:344]: `locals()` returns all locally defined variables (in this case all function parameters)

**Comment by noxan** [comment]
@karl-richter thanks for the notice, i'll address the pyright warnings elsewhere as they came from another PR

**Pull request merged**